src/server/api/pagination/find-many.proxy.ts

```diff
@@ -1,59 +0,0 @@
-import type { PaginationData, ProxyFunctions } from "./types";
-
-interface Paginated<T> {
-  data: T[];
-  count: number;
-  total: number | null;
-  cursor: string | null;
-}
-
-/**
- * FindManyPaginated
- *
- * type of the findManyPaginated method
- */
-export type FindManyPaginated<F extends ProxyFunctions> = {
-  findManyPaginated: (
-    data: Omit<Parameters<F["findMany"]>[0], "take" | "skip" | "cursor">,
-    pagination: PaginationData,
-  ) => Promise<Paginated<Awaited<ReturnType<F["findMany"]>>[0]>>;
-};
-
-/**
- * makeFindManyPaginated
- *
- * factory function that creates the findManyPaginated method.
- * this method is used to paginate the results of a findMany method.
- * this method implements js proxy to intercept the call to findMany and add the pagination logic.
- */
-export function makeFindManyPaginated(model: ProxyFunctions) {
```

Comment:
replaced this with prisma-pagination-extension which adds in prisma level. which applies to all model and no need to add proxy for each query